apps/etterlatte-saksbehandling-ui/client/src/components/behandling/vilkaarsvurdering/Vilkaarsvurdering.tsx

```diff
@@ -67,34 +41,36 @@ export const Vilkaarsvurdering = () => {
         </HeadingWrapper>
       </ContentHeader>
 
-      {behandlingId && status === RequestStatus.ok && vilkaarsvurdering?.virkningstidspunkt && (
+      {behandlingId && isSuccess(vilkaarsvurderingStatus) && vilkaarsvurderingStatus.data.virkningstidspunkt && (
         <>
           <VilkaarBorderTop />
-          {vilkaarsvurdering.vilkaar.map((value, index) => (
+          {vilkaarsvurderingStatus.data.vilkaar.map((value, index) => (
             <ManueltVilkaar
               key={index}
               vilkaar={value}
-              oppdaterVilkaar={oppdaterVilkaarsvurdering}
+              oppdaterVilkaar={(vilkaarsvurdering) => dispatch(updateVilkaarsvurdering(vilkaarsvurdering))}
               behandlingId={behandlingId}
-              redigerbar={behandles && !vilkaarsvurdering?.resultat}
+              redigerbar={behandles && !vilkaarsvurderingStatus.data?.resultat}
             />
           ))}
+          {vilkaarsvurderingStatus.data.vilkaar.length === 0 && <p>Du har ingen vilkår</p>}
 
           <Resultat
-            virkningstidspunktDato={vilkaarsvurdering.virkningstidspunkt}
-            vilkaarsvurdering={vilkaarsvurdering}
-            oppdaterVilkaar={oppdaterVilkaarsvurdering}
+            virkningstidspunktDato={vilkaarsvurderingStatus.data.virkningstidspunkt}
+            vilkaarsvurdering={vilkaarsvurderingStatus.data}
+            oppdaterVilkaar={(vilkaarsvurdering) => dispatch(updateVilkaarsvurdering(vilkaarsvurdering))}
             behandlingId={behandlingId}
             redigerbar={behandles}
           />
         </>
       )}
-      {vilkaarsvurdering?.vilkaar.length === 0 && <p>Du har ingen vilkår</p>}
-      {status === RequestStatus.isloading && <Spinner visible={true} label={'Henter vilkårsvurdering'} />}
-      {status === RequestStatus.error && <p>En feil har oppstått</p>}
-      {status === RequestStatus.preconditionFailed && (
-        <p>Virkningstidspunkt må avklares før vilkårsvurdering kan starte</p>
-      )}
+      {isPending(vilkaarsvurderingStatus) && <Spinner visible={true} label={'Henter vilkårsvurdering'} />}
+      {isFailure(vilkaarsvurderingStatus) &&
+        (vilkaarsvurderingStatus.error.statusCode === 412 ? (
```

Comment:
Kan også være statussjekken som feiler?

Reply:
Foreslår at du bare endrer status hvis statussjekken feiler

Reply:
what do you mean

Reply:
At vilkårsvurdering returnerer 412 for to ulike caser

Reply:
Ja, det er sant, kan gjøre den mer riktig 👍